spark_log_parser/loaders/https.py

```diff
@@ -29,13 +29,13 @@ def load_item(self, url):
         yield from self.extract(Path(url), stream)
 
 
-class HTTPFileBlobDataLoader(AbstractHTTPFileDataLoader, BlobFileReaderMixin):
+class HTTPFileBlobDataLoader(BlobFileReaderMixin, AbstractHTTPFileDataLoader):
```

Comment:
As you already mentioned, this does indeed work when changing the order of the classes here. I am not 100% sure how I feel about this being order-dependent, but, I guess this is just how Python handles the multiple inheritance "diamond problem" - https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem

> [Perl](https://en.wikipedia.org/wiki/Perl) uses the list of classes to inherit from as an ordered list. The compiler uses the first method it finds by [depth-first searching](https://en.wikipedia.org/wiki/Depth-first_search) of the superclass list or using the [C3 linearization](https://en.wikipedia.org/wiki/C3_linearization) of the class hierarchy. Various extensions provide alternative class composition schemes. The order of inheritance affects the class semantics. In the above ambiguity, class B and its ancestors would be checked before class C and its ancestors, so the method in A would be inherited through B. This is shared with [Io](https://en.wikipedia.org/wiki/Io_(programming_language)) and [Picolisp](https://en.wikipedia.org/wiki/Picolisp). In Perl, this behavior can be overridden using the mro or other modules to use [C3 linearization](https://en.wikipedia.org/wiki/C3_linearization) or other algorithms.[[9]](https://en.wikipedia.org/wiki/Multiple_inheritance#cite_note-9)

> [Python](https://en.wikipedia.org/wiki/Python_(programming_language)) has the same structure as Perl, but, unlike Perl, includes it in the syntax of the language. The order of inheritance affects the class semantics. Python had to deal with this upon the introduction of new-style classes, all of which have a common ancestor, object. Python creates a list of classes using the [C3 linearization](https://en.wikipedia.org/wiki/C3_linearization) (or Method Resolution Order (MRO)) algorithm. That algorithm enforces two constraints: children precede their parents and if a class inherits from multiple classes, they are kept in the order specified in the tuple of base classes (however in this case, some classes high in the inheritance graph may precede classes lower in the graph[[10]](https://en.wikipedia.org/wiki/Multiple_inheritance#cite_note-10)). Thus, the method resolution order is: D, B, C, A.[[11]](https://en.wikipedia.org/wiki/Multiple_inheritance#cite_note-11)

Reply:
And I agree that it does indeed make for a cleaner inheritance hierarchy!

Reply:
Yeah, I didn't expect to have to be aware of how Python resolves methods and how it's based in the syntax. It's been a fun adventure!

Reply:
Haha, agreed!